src/components/Assets/Assets.js

```diff
@@ -0,0 +1,246 @@
+      this.setState({
+        assetOptions: data.signalOptions,
+        loading: false, // Keep naming of options from imported helper
+      });
+    });
+  }
+
+  handleAutocompleteChange = e => {
+    e.persist();
+    this.setState({ typedAsset: e.target.value });
+  };
+
+  onAssetSelect = (value, item) => {
+    this.setState({ loading: true });
+    getAllAssetDetails(item)
+      .then(data => {
+        this.setState({
+          assetWorkOrdersData: data.workOrdersResponse.data.records,
+          assetServiceRequestsData: data.serviceRequestsResponse.data.records,
+          assetDetailsData: data.detailsResponse.data,
+          assetCamerasData: data.camerasResponse.data.records,
+          assetPreventativeMaintenanceData:
+            data.preventativeMaintResponse.data.records,
+          assetMapData: data.mapResponse.data,
+          assetDetectorsData: data.detectorsResponse.data.records,
+          assetSignalPriorityData: data.signalPriorityResponse.data.records,
+          assetPoleAttachmentsData: data.poleAttachmentsResponse.data.records,
+          assetTravelSensorData: data.travelSensorResponse.data.records,
+          assetApsButtonRequestsData:
+            data.apsButtonRequestsResponse.data.records,
+          assetCadStatusData: data.cadStatusResponse.data,
+        });
+      })
+      .then(() => {
+        this.setState({
+          selectedAsset: item.identifier,
+          typedAsset: item.identifier,
+          loading: false,
+        });
+      });
+  };
+
+  clearAssetSearch = () => {
+    this.setState({ selectedAsset: "", typedAsset: "" });
+  };
+
+  render() {
+    return (
+      <div>
+        <h1>
+          <FontAwesomeIcon icon={faMapMarkerAlt} /> Assets
+        </h1>
+
+        {this.state.assetOptions.length > 0 && (
+          <form>
+            <div className="form-group">
+              <label htmlFor={"asset"}>{"Search Assets"}</label>
+              <br />
+              <Autocomplete
+                getItemValue={item => item.id}
+                items={this.state.assetOptions}
+                inputProps={this.inputProps("asset")}
+                wrapperStyle={this.wrapperStyle}
+                menuStyle={this.menuStyle}
+                renderItem={(item, isHighlighted) =>
+                  this.renderItem(item, isHighlighted)
+                }
+                shouldItemRender={(item, value) =>
+                  this.shouldItemRender(item, value)
+                }
+                value={this.state.typedAsset}
+                onChange={this.handleAutocompleteChange}
+                onSelect={(value, item) => this.onAssetSelect(value, item)}
+              />
+              <button
+                type="button"
+                className="btn btn-danger ml-2 btn-lg"
+                onClick={this.clearAssetSearch}
+              >
+                Clear
+              </button>
+            </div>
+          </form>
+        )}
+
+        {this.state.loading ? (
```

Comment:
You could use the `&&` syntax here instead of the ternary. Either works...

Reply:
Nice, just changed this. Thanks for catching that.